Source/PersonalityInsightsV3/PersonalityInsights.swift

```diff
@@ -0,0 +1,274 @@
+        A raw sampling error for each characteristic is also returned.
+     - parameter consumptionPreferences: If true, then information inferred about consumption preferences is returned in addition to the results.
+     - parameter failure: A function executed if an error occurs.
+     - parameter success: A function executed with the personality profile.
+     */
+    public func getProfile(
+        version: String,
+        fromHTML html: String,
+        acceptLanguage: String? = nil,
+        contentLanguage: String? = nil,
+        rawScores: Bool? = nil,
+        consumptionPreferences: Bool? = nil,
+        failure: ((Error) -> Void)? = nil,
+        success: @escaping (Profile) -> Void)
+    {
+        guard let content = html.data(using: String.Encoding.utf8) else {
+            let failureReason = "HTML could not be encoded to NSData with NSUTF8StringEncoding."
+            let userInfo = [NSLocalizedFailureReasonErrorKey: failureReason]
+            let error = NSError(domain: domain, code: 0, userInfo: userInfo)
+            failure?(error)
+            return
+        }
+
+        getProfile(
+            version: version,
+            fromContent: content,
+            withType: "text/html",
+            acceptLanguage: acceptLanguage,
+            contentLanguage: contentLanguage,
+            rawScores: rawScores,
+            consumptionPreferences: consumptionPreferences,
+            failure: failure,
+            success: success
+        )
+    }
+
+    /**
+     Analyze input content items to generate a personality profile.
+     
+     - parameter version: The date in the form "YYYY-MM-DD" that will specify a version no later than the date provided.
+     - parameter contentItems: The content items to analyze.
+     - parameter acceptLanguage: The desired language of the response.
+     - parameter contentLanguage: The language of the text being analyzed.
+     - parameter rawScores: If true, then a raw score for each characteristic is returned in
+        addition to a normalized score. Raw scores are not compared with a sample population.
+        A raw sampling error for each characteristic is also returned.
+     - parameter consumptionPreferences: If true, then information inferred about consumption preferences is returned in addition to the results.
+     - parameter failure: A function executed if an error occurs.
+     - parameter success: A function executed with the personality profile.
+     */
+    public func getProfile(
+        version: String,
+        fromContentItems contentItems: [ContentItem],
+        acceptLanguage: String? = nil,
+        contentLanguage: String? = nil,
+        rawScores: Bool? = nil,
+        consumptionPreferences: Bool? = nil,
+        failure: ((Error) -> Void)? = nil,
+        success: @escaping (Profile) -> Void)
+    {
+        let json = JSON(dictionary: ["contentItems": contentItems.map { $0.toJSONObject() }])
```

Comment:
While reading through this code I realized just how poorly named the JSON functions are. I didn't understand at first glance why this line calls `toJSONObject()` instead of `toJSON`. (I remember now that it's because `toJSONObject()` returns `Any` instead of `JSON`...but it's all very confusing.)

Do you think we should rename some of the JSON functions in the future? Just curious to know what your thoughts are here.